packages/cli/config/jest.js

```diff
@@ -96,11 +99,61 @@ async function getConfig() {
   };
 
   // Use src/setupTests.ts as the default location for configuring test env
-  if (fs.existsSync('src/setupTests.ts')) {
+  if (fs.existsSync(path.resolve(targetPath, 'src/setupTests.ts'))) {
     options.setupFilesAfterEnv = ['<rootDir>/setupTests.ts'];
   }
 
   return Object.assign(options, ...pkgJsonConfigs);
 }
 
-module.exports = getConfig();
+// This loads the root jest config, which in turn will either refers to a single
+// configuration for the current package, or a collection of configurations for
+// the target workspace packages
+async function getRootConfig() {
+  const targetPath = process.cwd();
+  const targetPackagePath = path.resolve(targetPath, 'package.json');
+  const exists = await fs.pathExists(targetPackagePath);
+
+  if (!exists) {
+    return getProjectConfig(targetPath);
+  }
+
+  // Check whether the current package is a workspace root or not
+  const data = await fs.readJson(targetPackagePath);
+  const workspacePatterns = data.workspaces && data.workspaces.packages;
+  if (!workspacePatterns) {
+    return getProjectConfig(targetPath);
+  }
+
+  // If the target package is a workspace root, we find all packages in the
+  // workspace and load those in as separate jest projects instead.
+  const projectPaths = await Promise.all(
+    workspacePatterns.map(pattern => glob(path.join(targetPath, pattern))),
+  ).then(_ => _.flat());
+
+  const configs = await Promise.all(
+    projectPaths.flat().map(async projectPath => {
+      const packagePath = path.resolve(projectPath, 'package.json');
+      if (!(await fs.pathExists(packagePath))) {
+        return undefined;
+      }
+
+      // We check for the presence of "backstage-cli test" in the package test
```

Comment:
and we can't support other runners? is this just because it's the only way we can be sure that it's even jest based?

Reply:
It's depends quite a bit on our default test setup, so if you stray too far away from that you'd need to set something like this up yourself instead.